server/src/main/java/io/druid/server/ClientInfoResource.java

```diff
@@ -83,13 +100,83 @@ public ClientInfoResource(
   @Produces(MediaType.APPLICATION_JSON)
   public Map<String, Object> getDatasource(
```

Comment:
Is it not possible to throw an exception out of a Jersey resource and get Jersey to interpret that as an error response?